presto-main/src/main/java/com/facebook/presto/memory/AbstractAggregatedMemoryContext.java

```diff
@@ -26,11 +28,21 @@
 
     public AggregatedMemoryContext newAggregatedMemoryContext()
     {
-        return new AggregatedMemoryContext(this);
+        return childContextSupplier().get();
+    }
+
+    public Supplier<AggregatedMemoryContext> childContextSupplier()
+    {
+        return () -> new AggregatedMemoryContext(this);
     }
 
     public LocalMemoryContext newLocalMemoryContext()
     {
-        return new LocalMemoryContext(this);
+        return localContextSupplier().get();
+    }
+
+    public Supplier<LocalMemoryContext> localContextSupplier()
```

Comment:
Why do we need `LocalMemoryContext` and `AggregatedMemoryContext`. Why not merge these two?